pkg/utils/Idpool.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
+
+// Package linuxgeneralmodule is the main package of the application
+
+package utils
+
+import (
+	"log"
+	"reflect"
+	// "github.com/opiproject/opi-evpn-bridge/pkg/vendor_plugins/intel-e2000/p4runtime/p4translation"
+)
+
+// IDPool structure
+/*  Helper class for uniquely assigning IDs from a specified integer set (e.g. a
+#   range) to keys. IDs are assigned (or read) with get_id(key) and returned back
+#   into the pool with release_id(key). The IDPool remembers a once-assigned ID
+#   for keys so that the same ID is assigned for a key. Only when the pool runs
```

Comment:
why do we need such convoluted logic? Why we cannot pick any free id?

Reply:
The crucial feature of this ID allocator is that a unique DB object (as defined by its key fields) always gets the same ID assigned, no matter how often we request an ID. For the netlink watcher, we request an ID for the same objects every scanning interval. If it got completely independent IDs every time, it would be impossible to compare two subsequent Netlink DB shapshots and calculate the delta for generating add/modify/delete notifications.